Source/WebCore/platform/graphics/gtk/GdkSkiaUtilities.cpp

```diff
@@ -73,6 +73,18 @@ RefPtr<cairo_surface_t> skiaImageToCairoSurface(SkImage& image)
 }
 #endif
 
+GRefPtr<GdkPixbuf> skiaImageToGdkPixbuf(SkImage& image)
+{
+    SkPixmap pixmap;
+    if (!image.peekPixels(&pixmap))
+        return { };
+
+    // FIXME: This only supports RGB{A,X}_8888, so we could convert the SkImage if it was in a different format.
```

Comment:
I would add a check here as well, and return an empty image. Better safe than crashy!

```suggestion
    // FIXME: This only supports RGB{A,X}_8888, so we could convert the SkImage if it was in a different format.
    if (image.colorType() != SkColorType::kRGBA_8888_SkColorType && image.colorType() != SkColorType::kRGB_888x_SkColorType)
        return { };
```